dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -567,47 +529,23 @@ def validate_job(self, request, interfaces):
             job_type_name = job.job_type.name
         else:
             validation_error_type = ValidationError.jobError
-            writeFileError(job_id, self.filename, validation_error_type)
+            writeFileError(job_id, None, validation_error_type)
             raise ResponseException(
                 'Job ID {} is not a validation job (job type is {})'.format(job_id, job.job_type.name),
                 StatusCode.CLIENT_ERROR, None,
                 validation_error_type)
 
-        # todo: remove the following try/catch once 1st batch of changes are merged
-        try:
-            jobTracker.markJobStatus(job_id, "running")
-            if job_type_name == 'csv_record_validation':
-                self.runValidation(job_id, interfaces)
-            elif job_type_name == 'validation':
-                self.runCrossValidation(job_id, interfaces)
-            else:
-                raise ResponseException("Bad job type for validator",
-                    StatusCode.INTERNAL_ERROR)
-
-            return JsonResponse.create(StatusCode.OK, {"message":"Validation complete"})
-        except ResponseException as e:
-            _exception_logger.exception(str(e))
-            self.markJob(job_id, jobTracker, "invalid", self.filename,e.errorType, e.extraInfo)
-            return JsonResponse.error(e, e.status)
-        except ValueError as e:
-            _exception_logger.exception(str(e))
-            # Problem with CSV headers
-            exc = ResponseException(str(e),StatusCode.CLIENT_ERROR,type(e), ValidationError.unknownError) #"Internal value error"
-            self.markJob(job_id,jobTracker, "invalid", self.filename, ValidationError.unknownError)
-            return JsonResponse.error(exc, exc.status)
-        except Error as e:
```

Comment:
Thanks for the sharp eyes here..exactly what's needed. I definitely agonized a bit over all of this when stripping out this `try/except`. What I think is happening:

* The csv-related validation errors (missing/invalid headers, etc.) are being raised as a `ValueError` "wrappedException" within a `ResponseException`.  [Example](https://github.com/fedspendingtransparency/data-act-broker-backend/blob/development/dataactvalidator/filestreaming/csvAbstractReader.py#L48)
* Admittedly, I'm not sure what the purpose of the 'wrappedException' is, except to provide more information for debugging. It doesn't seem to affect the error type checking itself.
* Thus, these errors are caught as a `ResponseException` and not a `ValueError` and therefore handled by `handle_response_exception` in app.py, which sets the job status to `invalid`

To confirm, I ran a validation on a headerless csv, and the job status was invalid.

If you want to dig more, here's some before and after output when trying to validate a file with no header row:

*before*
```
2016-11-17 15:37:35,030 INFO:deprecated.info:Validator response: {'wrappedType': "<class 'ValueError'>", 'errorType': "<class 'dataactcore.utils.responseException.ResponseException'>", 'message': 'Errors in header row: Missing: adjustmentstounobligatedbalancebroughtforward_cpe, agencyidentifier, allocationtransferagencyidentifier, availabilitytypecode, beginningperiodofavailability, borrowingauthorityamounttotal_cpe, budgetauthorityappropriatedamount_cpe, budgetauthorityavailableamounttotal_cpe, budgetauthorityunobligatedbalancebroughtforward_fyb, contractauthorityamounttotal_cpe, deobligationsrecoveriesrefundsbytas_cpe, endingperiodofavailability, grossoutlayamountbytas_cpe, mainaccountcode, obligationsincurredtotalbytas_cpe, otherbudgetaryresourcesamount_cpe, spendingauthorityfromoffsettingcollectionsamounttotal_cpe, statusofbudgetaryresourcestotal_cpe, subaccountcode, unobligatedbalance_cpe', 'trace': ["('/Users/rebeccasweger/Dev/data-act-broker/dataactvalidator/validation_handlers/validationManager.py', 580, 'validate_job', 'self.runValidation(job_id, interfaces)')", "('/Users/rebeccasweger/Dev/data-act-broker/dataactvalidator/validation_handlers/validationManager.py', 308, 'runValidation', 'bucketName, errorFileName, self.long_to_short_dict)')", "('/Users/rebeccasweger/Dev/data-act-broker/dataactvalidator/filestreaming/csvLocalReader.py', 27, 'open_file', 'region, bucket, filename, csv_schema, bucket_name, error_filename, long_to_short_dict)')", '(\'/Users/rebeccasweger/Dev/data-act-broker/dataactvalidator/filestreaming/csvAbstractReader.py\', 133, \'open_file\', \'raise ResponseException("Errors in header row: " + str(error_string), StatusCode.CLIENT_ERROR, ValueError,ValidationError.headerError,**extra_info)\')'], 'wrappedMessage': 'Errors in header row: Missing: adjustmentstounobligatedbalancebroughtforward_cpe, agencyidentifier, allocationtransferagencyidentifier, availabilitytypecode, beginningperiodofavailability, borrowingauthorityamounttotal_cpe, budgetauthorityappropriatedamount_cpe, budgetauthorityavailableamounttotal_cpe, budgetauthorityunobligatedbalancebroughtforward_fyb, contractauthorityamounttotal_cpe, deobligationsrecoveriesrefundsbytas_cpe, endingperiodofavailability, grossoutlayamountbytas_cpe, mainaccountcode, obligationsincurredtotalbytas_cpe, otherbudgetaryresourcesamount_cpe, spendingauthorityfromoffsettingcollectionsamounttotal_cpe, statusofbudgetaryresourcestotal_cpe, subaccountcode, unobligatedbalance_cpe'}
```
*after*

```
2016-11-17 15:26:02,343 INFO:deprecated.info:Validator response: {'errorType': "<class 'dataactcore.utils.responseException.ResponseException'>", 'trace': ["('/Users/rebeccasweger/Dev/.virtualenvs/data-act-broker/lib/python3.4/site-packages/flask/app.py', 1639, 'full_dispatch_request', 'rv = self.dispatch_request()')", "('/Users/rebeccasweger/Dev/.virtualenvs/data-act-broker/lib/python3.4/site-packages/flask/app.py', 1625, 'dispatch_request', 'return self.view_functions[rule.endpoint](**req.view_args)')", "('dataactvalidator/app.py', 92, 'validate', 'return validation_manager.validate_job(request,interfaces)')", "('/Users/rebeccasweger/Dev/data-act-broker/dataactvalidator/validation_handlers/validationManager.py', 541, 'validate_job', 'self.runValidation(job, interfaces)')", "('/Users/rebeccasweger/Dev/data-act-broker/dataactvalidator/validation_handlers/validationManager.py', 269, 'runValidation', 'bucketName, errorFileName, self.long_to_short_dict)')", "('/Users/rebeccasweger/Dev/data-act-broker/dataactvalidator/filestreaming/csvLocalReader.py', 27, 'open_file', 'region, bucket, filename, csv_schema, bucket_name, error_filename, long_to_short_dict)')", '(\'/Users/rebeccasweger/Dev/data-act-broker/dataactvalidator/filestreaming/csvAbstractReader.py\', 133, \'open_file\', \'raise ResponseException("Errors in header row: " + str(error_string), StatusCode.CLIENT_ERROR, ValueError,ValidationError.headerError,**extra_info)\')'], 'message': 'Errors in header row: Missing: adjustmentstounobligatedbalancebroughtforward_cpe, agencyidentifier, allocationtransferagencyidentifier, availabilitytypecode, beginningperiodofavailability, borrowingauthorityamounttotal_cpe, budgetauthorityappropriatedamount_cpe, budgetauthorityavailableamounttotal_cpe, budgetauthorityunobligatedbalancebroughtforward_fyb, contractauthorityamounttotal_cpe, deobligationsrecoveriesrefundsbytas_cpe, endingperiodofavailability, grossoutlayamountbytas_cpe, mainaccountcode, obligationsincurredtotalbytas_cpe, otherbudgetaryresourcesamount_cpe, spendingauthorityfromoffsettingcollectionsamounttotal_cpe, statusofbudgetaryresourcestotal_cpe, subaccountcode, unobligatedbalance_cpe', 'wrappedType': "<class 'ValueError'>", 'wrappedMessage': 'Errors in header row: Missing: adjustmentstounobligatedbalancebroughtforward_cpe, agencyidentifier, allocationtransferagencyidentifier, availabilitytypecode, beginningperiodofavailability, borrowingauthorityamounttotal_cpe, budgetauthorityappropriatedamount_cpe, budgetauthorityavailableamounttotal_cpe, budgetauthorityunobligatedbalancebroughtforward_fyb, contractauthorityamounttotal_cpe, deobligationsrecoveriesrefundsbytas_cpe, endingperiodofavailability, grossoutlayamountbytas_cpe, mainaccountcode, obligationsincurredtotalbytas_cpe, otherbudgetaryresourcesamount_cpe, spendingauthorityfromoffsettingcollectionsamounttotal_cpe, statusofbudgetaryresourcestotal_cpe, subaccountcode, unobligatedbalance_cpe'}
```